app/models/mission/MissionTable.scala

```diff
@@ -519,6 +519,17 @@ object MissionTable {
     missions.filter(_.userId === userId.toString).map(_.distanceProgress).sum.run.getOrElse(0F) * METERS_TO_MILES
   }
 
+  /**
+    * Gets distance audited by a user in their current mission, if it exists, in meters.
+    *
+    * @param userId
+    * @return
+    */
+  def getCurrentDistanceAudited(userId: UUID): Float = db.withSession { implicit session =>
+    val currentDistance = getCurrentMissionInRegion(userId, UserCurrentRegionTable.currentRegion(userId).get).get.distanceProgress.getOrElse(0F)
```

Comment:
You are making two assumptions here that you can't make:
1. `currentRegion(userId).get` assumes that the user has already been assigned a region, which you can't guarantee.
1. `getCurrentMissionInRegion(...).get` assumes that the user already has a mission created for this region, which _should_ always be the case, but it's not worth letting our code have the possibility of totally failing by banking on that.

Let me show you a fancy syntax that I think would accomplish what you need here...
```
val currentDistance: Option[Float] = for {
  currentRegion <- UserCurrentRegionTable.currentRegion(userId)
  currentMission <- getCurrentMissionInRegion(userId, currentRegion)
} yield {
  currentMission.distanceProgress.getOrElse(0F)
}
```

And I think this function should really return an `Option[Float]`, and the value is `None` if the user either does not have a current region assigned, or they don't have a mission in their assigned region.

If you want to learn more about Options in Scala, I think that [this](https://freecontent.manning.com/using-option-in-scala-part-1-introduction/) is a pretty good resource. [Part 3](https://freecontent.manning.com/using-option-in-scala-part-3-for-comprehension/) goes into the fancy for comprehension syntax I used there.

I'm also totally down to do a quick video call today where I can go over that syntax if you'd like.

Reply:
Thanks so much for the feedback! Are you open for a call right now?

Reply:
yep!